projects/packages/search/src/classic-search/class-classic-search.php

```diff
@@ -95,29 +95,29 @@ protected function __construct() {
 	}
 
 	/**
-	 * Returns the singleton of the class. Does not perform any instantiation.
-	 *
-	 * @return static The class singleton.
-	 */
-	public static function instance() {
-		return self::$instance;
-	}
-
-	/**
-	 * Instantiate and initialize a singleton instance of the class.
+	 * Returns a class singleton. Initializes with first-time setup if given a blog ID parameter.
 	 *
 	 * @param string $blog_id Blog id.
 	 * @return static The class singleton.
 	 */
-	public static function initialize( $blog_id ) {
+	public static function instance( $blog_id = null ) {
 		if ( ! isset( self::$instance ) ) {
+			if ( null === $blog_id ) {
+				$blog_id = Helper::get_wpcom_site_id();
+			}
 			self::$instance = new static();
 			self::$instance->setup( $blog_id );
 		}
-
 		return self::$instance;
 	}
 
+	/**
+	 * Alias of the instance function.
+	 */
+	public static function initialize() {
+		return call_user_func_array( array( 'self', 'instance' ), func_get_args() );
```

Comment:
This blows my mind, but works 👍  I thought `self` is a bit ambiguous here, but it works without any issues.